src/styles/index.scss

```diff
@@ -1,3 +1,90 @@
-body {
+* {
+  padding: 0;
   margin: 0;
+  box-sizing: border-box;
+}
+
+$size: 100px;
+$border: 1px solid black;
+$background: #eee;
+$text-color: #000;
+$text-size: 30px;
+$gap: 1px;
+$hover-color: #ffbfcb;
+$transition-duration: 0.5s;
+$width-calendar: 7 * $size + $gap * 6;
+$days: (
+  sun: 6,
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+);
+
+.block_calendar {
+  width: 100%;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  font-family: Arial, Helvetica, sans-serif;
+  display: flex;
+  flex-wrap: wrap;
+  width: $width-calendar;
+  gap: $gap;
+
+  @each $day, $index in $days {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: ($index * ($size + $gap));
+      }
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i}) {
+    width: $size;
+    height: $size;
+    background-color: $background;
+    border: $border;
+    cursor: pointer;
+    position: relative;
+
+    transition:
+      background-color $transition-duration,
+      transform $transition-duration;
+
+    &:hover {
+      background-color: $hover-color;
+      transform: translateY(-20px);
+    }
+
+    &::before {
+      content: '#{$i}';
+      font-size: $text-size;
+      color: $text-color;
+      position: absolute;
+      top: 50%;
+      left: 50%;
+      transform: translate(-50%, -50%);
+    }
+  }
+}
```

Comment:
The loop to generate `.calendar__day` styles is efficient, but it's generating styles for all 31 days regardless of the month length. It would be more efficient to generate only the necessary number of days within the `.calendar--month-length-#{$length}` loop (lines 80-90).